.github/workflows/build.yaml

```diff
@@ -0,0 +1,48 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+env:
+  go_version: 1.21.7  # Should match the value in arcalot/.github/.../go_lint_and_test.yaml
```

Comment:
[Can we do better than a "should match" comment?](https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows) 😦

Reply:
Yes, I think we can.

Thanks for the reference:  I'll look into that.  I didn't want the caller repo to have to make the decision -- I want it made centrally in _one place_, but I guess we can create organization variables for that.

Reply:
Done!  Although, it's proving challenging to test...but, I think it's working...mostly...but we probably won't know for sure until after this PR is merged.  🥴